### DISCUSSION TITLE
Help my esp no longer works

### DISCUSSION THREAD

**Discussion by PatLub85** [micropython/micropython#12606] [General]
Hi,
I have an esp8266 D1 mini which works well. I wanted to put the latest version ESP8266_GENERIC-20230426-v1.20.0 with Thonny 4.1.3.
Impossible to take control since this update. The LED flashes constantly and I get random messages on the console, and then Thonny freezes.
Is there a solution to get it working again?
Thanks in advance

**Comment by 2dof** [comment]
Have you used [esptool](https://randomnerdtutorials.com/flashing-micropython-firmware-esptool-py-esp32-esp8266/)  for flashing? 
I recommend to use esptool  but with just command prompt  or use [uPyCraft ide to  upload firmware](https://randomnerdtutorials.com/flash-upload-micropython-firmware-esp32-esp8266/)

**Reply by PatLub85** [reply]
Hi 2dof,
I followed your advice and it worked. On the other hand, I had to use an old PC because on my recent work PC (Windows 11) I was not able to install Upycraft, I got the message ''The application could not start correctly (0xc000007b) ''. Looking on the web, I saw that I was not the only one.
On the other hand, I ordered 2 ESP32 DevKitc v4. The first worked well, but with the second I got an error and it was impossible to erase the flash. Windows sees com port
An idea before I throw it in the trash

**Comment by robert-hh** [comment]
> The LED flashes constantly and I get random messages on the console, 

It looks as if the firmware was flashed using the wrong flash mode. AFAIK, the flash mode can be chosen even using Thonny. With esptool.py, it's the -fm option. Try again flashing the firmware with a different flash mode. There are 4 options, so it's easy to try all of them. There is no risk of damage if you use the wrong one.
